rmw_fastrtps_shared_cpp/src/rmw_init.cpp

```diff
@@ -62,39 +68,47 @@ rmw_init_options_copy(
     return RMW_RET_INVALID_ARGUMENT;
   }
```

Comment:
without line 66-69, this function could perfectly work with a non-zero initialized `dst` AFAIS

Reply:
I know, I prepared it for it. But it's lacking a finalization halfway through.

We can discuss guarantees later on and apply changes as needed.